metrics/src/counter.rs

```diff
@@ -76,7 +76,7 @@ macro_rules! inc_counter_info {
 #[macro_export]
 macro_rules! inc_new_counter {
     ($name:expr, $count:expr, $level:expr, $lograte:expr, $metricsrate:expr) => {{
-        if log_enabled!($level) {
+        if $count != 0 && log_enabled!($level) {
```

Comment:
nit: separate out little changes like this into their own PRs.  They can land independently, and sooner

Reply:
I actually reverted this change and instead put the `if ... != 0` at the call-site.
Though the counter value does not change when event count is zero, some of the logging logic in the implementation relies on how often that function is called regardless of events counts being zero or not:
https://github.com/solana-labs/solana/blob/f5ed017d6/metrics/src/counter.rs#L172-L203
It might be somewhere in the code relies on that behavior.